### PR TITLE
OCP Deployment on vSphere via Assisted Installer

### PR DESCRIPTION
**Pull request by dahorak**
This PR implements OCP deployment on vSphere via Assisted Installer.
It depends on [PR#8994](https://github.com/red-hat-storage/ocs-ci/pull/8994) and [PR#9042](https://github.com/red-hat-storage/ocs-ci/pull/9042) PRs.

### CODE REVIEW

**Comment by dahorak** [comment]
Verification job: https://url.corp.redhat.com/e7715e3 - OCP deployment passed.

**Review by shylesh — APPROVED** [review]
LGTM

**Review by vavuthu — APPROVED** [review]

File: conf/deployment/vsphere/ai_1az_rhcos_vsan_3m_3w_hci.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  allow_lower_instance_requirements: false
+ENV_DATA:
+  platform: 'vsphere'
+  deployment_type: 'ai'
+  worker_replicas: 3
+  master_replicas: 3
+  master_num_cpus: '16'
+  worker_num_cpus: '16'
+  master_memory: '65536'
+  compute_memory: '65536'
```
Comment (vavuthu) [inline conf/deployment/vsphere/ai_1az_rhcos_vsan_3m_3w_hci.yaml:13]: is it required for both master and worker has same CPU's and memory. Usally master will have less resource requirement than workers.

Comment (dahorak) [reply conf/deployment/vsphere/ai_1az_rhcos_vsan_3m_3w_hci.yaml:13]: This is the requirement for Provider/Client deployment (for the normal deployment there is the first configuration file `conf/deployment/vsphere/ai_1az_rhcos_vsan_3m_3w.yaml` with less CPU's and memory for master nodes).

**Review by vavuthu — DISMISSED** [review]

File: ocs_ci/deployment/vmware.py [hunk 2]
```diff
@@ -1575,6 +1576,302 @@ def destroy_cluster(self, log_level="DEBUG"):
...
+                "api_ip": self.api_vip,
+                "ingress_ip": self.ingress_vip,
+                "iso_image": self.discovery_iso_image,
+                "control_plane_count": config.ENV_DATA["master_replicas"],
+                "compute_count": config.ENV_DATA["worker_replicas"],
+                "control_plane_memory": config.ENV_DATA["master_memory"],
+                "compute_memory": config.ENV_DATA["compute_memory"],
+                "control_plane_num_cpus": config.ENV_DATA["master_num_cpus"],
+                "compute_num_cpus": config.ENV_DATA["worker_num_cpus"],
+                "system_disk_size": "120",
```
Comment (petr-balogh) [inline ocs_ci/deployment/vmware.py:1748]: https://github.com/red-hat-storage/ocs-ci/blob/cc1f7b09de400a9b026c023d3a968ed106886bde/ocs_ci/ocs/constants.py#L1281

**Review by petr-balogh — DISMISSED** [review]

**Review by DanielOsypenko — DISMISSED** [review]

File: conf/deployment/vsphere/ai_1az_rhcos_vsan_3m_3w.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# This config is suppose to work on DC-CP, DC-ECO, DC-PS
+DEPLOYMENT:
+  allow_lower_instance_requirements: false
+ENV_DATA:
+  platform: 'vsphere'
+  deployment_type: 'ai'
+  worker_replicas: 3
+  master_replicas: 3
+  worker_num_cpus: '16'
+  master_num_cpus: '4'
```
[the remark this answers is not in the document]
Comment (dahorak) [reply conf/deployment/vsphere/ai_1az_rhcos_vsan_3m_3w.yaml:10]: This config is meant for "normal" deployment (not used anywhere now). The `*_hci.yaml` config have the required number of CPUs and memory.

File: conf/deployment/vsphere/ai_1az_rhcos_vsan_3m_3w_hci.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+---
+# This config is suppose to work on DC-CP, DC-ECO, DC-PS
+DEPLOYMENT:
+  allow_lower_instance_requirements: false
```
[the remark this answers is not in the document]
Comment (dahorak) [reply conf/deployment/vsphere/ai_1az_rhcos_vsan_3m_3w_hci.yaml:4]: updated

**Review by petr-balogh — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Pull request merged**